OpenRA.Game/FileSystem/MixFile.cs

```diff
@@ -122,6 +122,18 @@ static MemoryStream Decrypt(uint[] h, Blowfish fish)
 
 		static uint[] ReadBlocks(Stream s, long offset, int count)
 		{
+			if (offset < 0)
+				throw new ArgumentOutOfRangeException("offset", "Non-negative number required.");
+
+			if (offset > s.Length)
+				throw new ArgumentOutOfRangeException("offset", "Offset {0} greater than stream length {1}.".F(offset, s.Length));
```

Comment:
Sorry I meant to imply we can do without this check, since the offset + bytes count you've now added checks the same thing (and more) anyway.